contracts/DomainController.sol

```diff
@@ -93,14 +96,16 @@ contract DomainController is Initializable, OwnableUpgradeable {
             revert InsufficientPayment();
```

Comment:
// line 95 
// this check is actually incorrect
// you try to check if the value is a string  if (msg.value < _getDomainStorage().domainRegistrationFee)
// but the value can be more than the fee
// but on the line 99 you add domain registration fee as reward for owner 
// so if registration fee === 1eth and user sends 2 eth, the owner will get 1 eth as reward
// and  1 eth will be stored in the contract
// and there is no way to get it back

// // - 5 points